twisted/names/test/test_rootresolve.py

```diff
@@ -209,20 +209,20 @@ def test_lookupChecksClass(self):
         badClass.CLASS = HS
         servers = {
             ('1.1.2.3', 53): {
-                ('foo.example.com', A): {
-                    'answers': [('foo.example.com', badClass)],
-                    'authority': [('foo.example.com', Record_NS('ns1.example.com'))],
-                    'additional': [('ns1.example.com', Record_A('10.0.0.2'))],
+                (b'foo.example.com', A): {
+                    'answers': [(b'foo.example.com', badClass)],
+                    'authority': [(b'foo.example.com', Record_NS(b'ns1.example.com'))],
+                    'additional': [(b'ns1.example.com', Record_A('10.0.0.2'))],
                 },
             },
             ('10.0.0.2', 53): {
-                ('foo.example.com', A): {
-                    'answers': [('foo.example.com', Record_A('10.0.0.3'))],
+                (b'foo.example.com', A): {
+                    'answers': [(b'foo.example.com', Record_A('10.0.0.3'))],
                 },
             },
         }
         resolver = self._getResolver(servers)
-        d = resolver.lookupAddress('foo.example.com')
+        d = resolver.lookupAddress(b'foo.example.com')
```

Comment:
The problem with expanding `IResolver` to take `unicode` is that valid, 3rd-party implementations of this interface may have taken it at face value and implemented only `bytes` support (since it only supported py2 previously).  So when passing values here we should probably be conservatively passing `bytes` only.

However, semantically, this is wrong - DNS resolves _names_, which are textual values.  So the correct long-term solution is to define a `IResolverButGooder` interface which does explicitly document that it takes `unicode`, then implement that.  See also https://twistedmatrix.com/trac/ticket/4362